core/node/state_keeper/src/io/mempool.rs

```diff
@@ -298,6 +298,31 @@ impl StateKeeperIO for MempoolIO {
                         .await?;
                     continue;
                 }
+
+                // Reject transactions that violate block.timestamp constraints. Such transactions should be
+                // rejected at the API level, but we need to protect ourselves in case if a transaction
+                // goes outside of the allowed range while being in the mempool
```

Comment:
Dumb question: The PR description says that changing these params doesn't require a protocol upgrade. How will params work in a decentralized network? How will these values be coordinated among validators? (I'd imagine the timestamp asserter contract would actually contain these values in this case.) Or is this too far too ahead to plan?

Reply:
We could in theory move these checks down to the `TimestampAsserter` contract and have those variables set there. @StanislavBreadless  would love your thoughts here.

Reply:
I think it is better not to do things on the protocol side that can be moved somewhere else. (In the worst case, consensus can be done offchain). 

The reason why communicating it offchain may make sense is because this restriction is on the API side only. I.e. if someone proposes a miniblock, the validators that actually check the block should not be affected by this restriction, i.e. if the validation of the custom account passes, the block is valid. Otherwise it is not. The restrictions here are just to protect the block builder and not the other validators